### PR TITLE
DOC New limit behaviour

### PR DESCRIPTION
**Pull request by maxime-rainville**
## Parent issue
https://github.com/silverstripe/silverstripe-framework/issues/10604

### CODE REVIEW

File: en/04_Changelogs/5.0.0.md [hunk 1]
```diff
@@ -205,6 +205,9 @@ This is a major release and as a result there are a number of breaking API chang
...
 - [DataList](SilverStripe\ORM\DataList), its subclasses, [Map](api:SilverStripe\ORM\Map), and [ArrayList](SilverStripe\ORM\ArrayList) all now return generators from `getIterator()`. This reduces memory usage when looping over large result sets. As a result of this, `getGenerator()` has been removed as it is no longer needed. Note that `chunkedFetch()` has not been removed, as it may still be useful for very large result sets to fetch results in smaller chunks at a time.
+- [Limitable::limit()](SilverStripe\ORM\Limitable::limit()) is now strongly typed. Calling `limit()` with a `0` or `false` length now limits the list to 0 entry. In Silverstripe CMS 4, any "falsy" value would unset the limit. To unset a list's limit in Silverstripe CMS 5 , call `limit()` with an explicit `null`. This affects [DataList](SilverStripe\ORM\DataList), [ArrayList](SilverStripe\ORM\ArrayList) and all other classes implementing Limitable](SilverStripe\ORM\Limitable)
+  - Calling [SQLSelect::setLimit()](SilverStripe\ORM\Queries\SQLSelect::setLimit()) with `0` used to unset the limit. It will now set a limit of `0` causing the query to return no results. `setLimit()` with `null` to unset the limit.
```
Comment (sabina-talipova) [inline en/04_Changelogs/5.0.0.md]:
```suggestion
  - In Silverstripe CMS 4, calling [SQLSelect::setLimit()](SilverStripe\ORM\Queries\SQLSelect::setLimit()) with `0` as limit argument was used to unset the limit. in Silverstripe CMS 5, set a limit of `0` will be causing the query to return no results. Now, to unset the limit use `setLimit()` with `null` as limit value.
```

Comment (sabina-talipova) [reply en/04_Changelogs/5.0.0.md]: Could we paraphrase this sentence to make it more accessible for understanding?

File: en/04_Changelogs/5.0.0.md [hunk 2]
```diff
@@ -205,6 +205,9 @@ This is a major release and as a result there are a number of breaking API chang
...
+- [ArrayList::limit()](SilverStripe\ORM\ArrayList::limit()) now throws an `InvalidArgumentException` when called with a negative `$length` or `$offset`.
```
Comment (sabina-talipova) [inline en/04_Changelogs/5.0.0.md]:
```suggestion
- [ArrayList::limit()](SilverStripe\ORM\ArrayList::limit()) in Silverstripe CMS 5 throws an `InvalidArgumentException` when called with a negative `$length` or `$offset`.
```

**Review by sabina-talipova — CHANGES_REQUESTED** [review]

**Comment by sabina-talipova** [comment]
All looks great, but could we add a little bit more details about versions.
Thank you.

File: en/04_Changelogs/5.0.0.md (hunk 1 above)
Comment (maxime-rainville) [reply en/04_Changelogs/5.0.0.md]:
```suggestion
  - In Silverstripe CMS 4, calling [SQLSelect::setLimit()](SilverStripe\ORM\Queries\SQLSelect::setLimit()) with `0` as argument would unset the limit. In Silverstripe CMS5, it sets a limit of `0` causing the query to return no results. Call `setLimit()` with `null` to unset the limit.
```

Comment (maxime-rainville) [reply en/04_Changelogs/5.0.0.md]: Tweaked your suggestion a bit, but otherwise kept most of it.

**Review by sabina-talipova — APPROVED** [review]

**Pull request merged**